compiler/utils/cli.ml

```diff
@@ -80,19 +80,61 @@ let backend =
     & info [] ~docv:"COMMAND"
         ~doc:"Backend selection (see the list of commands for available options).")
 
-type backend_option =
-  | Latex
-  | Makefile
-  | Html
-  | Interpret
-  | Typecheck
-  | OCaml
-  | Python
-  | Scalc
-  | Lcalc
-  | Dcalc
-  | Scopelang
-  | Proof
+let plugins_dirs =
+  let doc = "Set the given directory to be searched for backend plugins." in
+  let env = Term.env_info "CATALA_PLUGINS" ~doc in
+  let default =
+    let ( / ) = Filename.concat in
+    [ Sys.executable_name / ".." / "lib" / "catala" / "plugins" ]
+  in
+  Arg.(value & opt_all dir default & info [ "plugin-dir" ] ~docv:"DIR" ~env ~doc)
+
+type backend_option_builtin =
```

Comment:
Hi @AltGr, are there any advantages to using polymorphic variants rather than ordinary ones for this case?

Reply:
Yes, see the [individual commit messages](https://github.com/CatalaLang/catala/pull/213/commits/a212d529ad2d91f5617148e4744f2f8ee0ddc4fe):
- the last part is optional, I can roll it back if preferred: it makes use of
  polymorphic variants, which some find distasteful. `driver.ml` proceeds with
  successive refinements of the `Cli.backend_option` type, and in this case they
  fit well by allowing to type these refinements, ensuring both the absence of
  holes and redundance, while removing the need for pervasive `_ -> assert
  false` cases.

I can happily either remove this part (and restore the `assert false` cases) or move it to a separate PR for further discussion, if desired.

Reply:
> and in this case they fit well by allowing to type these refinements, ensuring both the absence of holes and redundance, while removing the need for pervasive _ -> assert false cases.

I understand the point, thanks!